### PR TITLE
Add manage search query rules permission

### PR DESCRIPTION
**Pull request by kderusso**
Adds a permission to manage search query rules.

### CODE REVIEW

**Comment by kderusso** [comment]
Relates to PR to manage synonyms: https://github.com/elastic/elasticsearch/pull/97853

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/DeleteQueryRulesetAction.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 public class DeleteQueryRulesetAction extends ActionType<AcknowledgedResponse> {
 
     public static final DeleteQueryRulesetAction INSTANCE = new DeleteQueryRulesetAction();
-    public static final String NAME = "cluster:admin/xpack/query_rules/delete";
+    public static final String NAME = "cluster:admin/search/query_rules/delete";
```
Comment (ioanatia) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/DeleteQueryRulesetAction.java:32]: why did we change this?

**Review by ioanatia — COMMENTED** [review]
should we have some yaml tests similar to what we had for https://github.com/elastic/elasticsearch/pull/97853 ?

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/DeleteQueryRulesetAction.java (hunk 1 above)
Comment (kderusso) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/DeleteQueryRulesetAction.java:32]: I renamed this based on the discussion we had to make a manage search privilege that these fall under. If we don't want to do this I can change it back to `xpack`. WDYT?

Comment (carlosdelest) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/DeleteQueryRulesetAction.java:32]: As mentioned in https://github.com/elastic/elasticsearch/pull/97853, I don't think that's necessary. I also don't see a common grouping like that in the rest of the actions. I'd vote for keeping as is unless we have a good reason not to.

**Comment by carlosdelest** [comment]
> should we have some yaml tests similar to what we had for https://github.com/elastic/elasticsearch/pull/97853 ?

++. We could probably achieve this by adding auth headers to all the existing tests, and adding an additional use case with a user with no permissions.

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/DeleteQueryRulesetAction.java (hunk 1 above)
Comment (kderusso) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/DeleteQueryRulesetAction.java:32]: OK, I'll change them back. My bad, I misunderstood the goal.

Comment (kderusso) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/DeleteQueryRulesetAction.java:32]: On second thought, should it be `xpack/search/query_rules`?

Comment (carlosdelest) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/DeleteQueryRulesetAction.java]: I'd still say that we don't need the /search/ prefix. I find them more descriptive without it. Not a strong opinion though.

**Review by ioanatia — APPROVED** [review]

**Review by carlosdelest — APPROVED** [review]
👏

**Pull request merged**